### PR TITLE
Parse static blocks with typescript plugin

### LINKED ISSUE #13234: [Bug]: `Unexpected token` syntax error is thrown with `classStaticBlock` and `typescript`.
### 💻

- [X] Would you like to work on a fix?

### How are you using Babel?

Other (Next.js, Gatsby, vue-cli, ...)

### Input code

```ts
class Foo {
  static {}
}
```

### Configuration file name

_No response_

### Configuration

Use `classStaticBlock` and `typescript` plugin.

```js
const { parse } = require("@babel/parser");

const code =`class Foo {
  static {}
}`;

const result = parse(code, {
  sourceType: "module",
  errorRecovery: true,
  plugins: ["classStaticBlock", "typescript"],
});
```

### Current and expected behavior

**current:**

SyntaxError: Unexpected token

**expected:**

no error

### Environment

- `@babel/parser` 7.14.0

### Possible solution

_No response_

### Additional context

https://github.com/prettier/prettier/pull/10802#discussion_r623549163

### PR DESCRIPTION
**Pull request by sosukesuzuki**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            |  Fixes #13234
| Patch: Bug Fix?          | Y
| Tests Added + Pass?      | Yes
| License                  | MIT

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13243"></a>

### CODE REVIEW

File: packages/babel-parser/src/plugins/typescript/index.js [hunk 1]
```diff
@@ -2333,18 +2333,18 @@ export default (superClass: Class<Parser>): Class<Parser> =>
...
-        this.parseClassMemberWithIsStatic(
-          classBody,
-          member,
-          state,
-          !!member.static,
-        );
+      const callParseClassMemberWithIsStatic = () => {
+        const isStatic = !!member.static;
+        if (isStatic && this.eat(tt.braceL)) {
+          this.parseClassStaticBlock(classBody, ((member: any): N.StaticBlock));
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/src/plugins/typescript/index.js:2339]: We might also need explicit checks to disallow things like
```js
class A {
  readonly static {}
  static public {}
}
```

File: packages/babel-parser/test/fixtures/typescript/static-blocks/invalid-static-block-with-accessibility-private/input.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class Foo {
+  static private {}
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/test/fixtures/typescript/static-blocks/invalid-static-block-with-accessibility-private/input.ts:2]: We can report a better error, like `Static class blocks cannot have any modifier`

File: packages/babel-parser/src/plugins/typescript/index.js [hunk 3]
```diff
@@ -131,6 +131,8 @@ const TSErrors = makeErrorTemplates(
       "A 'set' accessor cannot have rest parameter.",
     SetAccesorCannotHaveReturnType:
       "A 'set' accessor cannot have a return type annotation.",
+    StaticBlockCannotHaveanyModifier:
```
Comment (JLHwung) [inline packages/babel-parser/src/plugins/typescript/index.js:134]:
```suggestion
    StaticBlockCannotHaveModifier:
```
nit-picking: I prefer shorter error codes.

File: packages/babel-parser/test/fixtures/typescript/static-blocks/static-blocks-and-static-property-named-static/input.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+class Foo {
+  static foo;
```
Comment (JLHwung) [inline packages/babel-parser/test/fixtures/typescript/static-blocks/static-blocks-and-static-property-named-static/input.ts:2]: As I read the name of the test, did you mean
```suggestion
  static static;
```
?

**Review by JLHwung — APPROVED** [review]
LGTM with some comments.

**Review by JLHwung — COMMENTED** [review]
Can you add a test case on `static static {}`? I think it should throw static block can not have static modifier.

**Comment by sosukesuzuki** [comment]
Sorry I pushed from another laptop, then my commit date is strange.., I'll fix

**Review by nicolo-ribaudo — APPROVED** [review]

**Pull request merged**